### PR TITLE
Expose more functions

### PR DESCRIPTION
**Pull request by skarupke**
- arg_min / arg_max
- ceil/floor
- clip_min/clip_max
- cache
- is_nan/is_not_nan
- set_sorted_flag

### CODE REVIEW

**Comment by skarupke** [comment]
Also added Expr.take and the constructor for Series literal now.

File: test/expressions.ml [hunk 1]
```diff
@@ -111,6 +111,21 @@ let%expect_test "Functions" =
     │ null ┆ egg   ┆ 0.156988 ┆ C      │
     │ 5    ┆ spam  ┆ 0.831802 ┆ B      │
     └──────┴───────┴──────────┴────────┘ |}];
+  let min_max = Data_frame.select_exn df ~exprs:
```
Comment (mt-caret) [inline test/expressions.ml:114]: Sorry, the tests don't make this super clear but the tests I see you've changed unfortunately are to the ones that are supposed to correspond to the Polars user guide (see https://github.com/mt-caret/polars-ocaml/pull/28 where we're going to extract them out into a separate directory).

Could we put these expect tests into a new file? I think eventually they belong in doctests in the .mli but that PR (https://github.com/mt-caret/polars-ocaml/pull/37) is blocked on a fairly large number of other PRs.

File: test/expressions.ml [hunk 2]
```diff
@@ -1376,3 +1441,83 @@ let%expect_test "Lists and Arrays" =
     │ Station 10 ┆ 17    ┆ 13    ┆ 10    ┆ [0.33, 0.67, 1.0]  │
     └────────────┴───────┴───────┴───────┴────────────────────┘ |}]
 ;;
+
+let%expect_test "join_asof" =
```
Comment (mt-caret) [inline test/expressions.ml:1445]: Have you formatted this code with ocamlformat? If not, please install it and run `dune build @fmt` which should apply formatting correctly.

File: test/expressions.ml [hunk 3]
```diff
@@ -1376,3 +1441,83 @@ let%expect_test "Lists and Arrays" =
...
+;;
+
+let%expect_test "take" =
+  let values = Series.float "val" [ 1.; 2.; 3.; 4.; 5.; 6. ] in
+  let idxs = Series.int "idx"   [ 1;  0;  1;  2;  0;  2 ] in
+  let ldf = Data_frame.create_exn[ values; idxs ] |> Data_frame.lazy_ in
+  let filtered = Lazy_frame.filter ldf ~predicate:(
+      (* Remove index 0, keep index 1 and 2 *)
+      Expr.series (Series.bool "filter" [ false; true; true ])
+      |> Expr.take ~idx:(Expr.col "idx"))
```
Comment (mt-caret) [inline test/expressions.ml:1507]: Could we do something like `~predicate:Expr.( ... )` so we can reduce the number of mentions of `Expr`?

**Review by mt-caret — CHANGES_REQUESTED** [review]
Thanks so much for the PR; I have a few comments but it otherwise looks fine to me!

File: lib/series.ml [hunk 4]
```diff
@@ -71,6 +71,13 @@ module T = struct
 
   let sort ?(descending = false) t = sort t ~descending
 
+  module IsSorted = struct
+      type t =
+        | Ascending
+        | Descending
+        | Not
+  end
```
[the remark this answers is not in the document]
Comment (skarupke) [reply lib/series.ml]: Done

File: test/expressions.ml (hunk 1 above)
Comment (skarupke) [reply test/expressions.ml]: I moved it to a separate file, "expression_test.ml". The name is almost a duplicate now, but that won't be a problem after the other tests are in a separate directory.

File: test/expressions.ml (hunk 2 above)
Comment (skarupke) [reply test/expressions.ml]: I had not formatted the code. Did that now.

File: test/expressions.ml (hunk 3 above)
Comment (skarupke) [reply test/expressions.ml]: Done

**Review by mt-caret — APPROVED** [review]

**Pull request merged**